packages/block-library/src/navigation/index.js

```diff
@@ -52,6 +54,25 @@ export const settings = {
 	},
 	edit,
 	save,
+	__experimentalLabel: ( { ref } ) => {
+		if ( ! ref ) {
+			return __( 'Navigation' );
+		}
+
+		const navigation = select( coreStore ).getEntityRecord(
+			'postType',
+			'wp_navigation',
+			ref
+		);
+
+		return navigation?.title?.rendered
+			? sprintf(
+					/* translators: %s: menu title */
+					__( 'Navigation (%s)' ),
+					navigation.title.rendered
```

Comment:
This is great :) 

We need to ensure that any encoded HTML entities in the title are decoded before rendering it.

See https://github.com/WordPress/gutenberg/blob/e746a95613db3eade8340c940cedbbd2b10145d6/packages/block-library/src/block/index.js#L30-L39